### PR TITLE
Remove :runtime_banner feature flag

### PR DESCRIPTION
**Pull request by fdoxyz**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

This PR removes the `runtime_banner` FeatureFlag, making the banner visible to all iOS users (Android or Desktop browsers shouldn't be affected by this).

The PR also makes a small adjustment to the positioning of the Banner: Making it slightly bigger and positioned lower on the screen.

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/72

## QA Instructions, Screenshots, Recordings

- Boot the app locally and browse it using an iOS device
   - The banner should be visible only to iOS devices and not other platforms
- You can also remove the `Browser-iOS` class from the banner itself (in `app/views/layouts/application.html.erb`)
   - Then you can use the DevTools to browse as if using a mobile device from your desktop browser

#### Before (positioning)

Screen Shot 2021-05-05 at 06 21 09

#### After (positioning)

Screen Shot 2021-05-27 at 17 12 37

### UI accessibility concerns?

None because it's not modifying the existing elements

## Added tests?

- [x] No, and this is why: Tests already exist and they are only modified to work without the FeatureFlag

## [Forem core team only] How will this change be communicated?

- [x] I will share this change internally with the appropriate teams
- [x] This change does not need to be communicated, and this is why not: This is part of the Forem iOS app release and will be communicated with the help of the Marketing team

## [optional] Are there any post deployment tasks we need to perform?

None

## [optional] What gif best describes this PR or how it makes you feel?

human billboard

### CODE REVIEW

File: app/assets/javascripts/initializers/initializeRuntimeBanner.js
Comment (fdoxyz) [inline app/assets/javascripts/initializers/initializeRuntimeBanner.js:59]: I _think_ this will be our AppStore URL but we just need to double check this before merging (as soon as the app is officially published).

**Review by Ridhwana — APPROVED** [review]
This looks good to me. 🎉

File: app/views/layouts/application.html.erb
Comment (vaidehijoshi) [inline app/views/layouts/application.html.erb:107]: Interesting...is `Browser-iOS` a class that comes from somewhere else? It just caught my eye because it's formatted so oddly (not like usual CSS) so I was wondering if it's a class that comes from/is defined elsewhere

**Review by vaidehijoshi — APPROVED** [review]

File: app/views/layouts/application.html.erb
Comment (citizen428) [reply app/views/layouts/application.html.erb:107]: `app/assets/javascripts/initializers/runtime.js` see https://github.com/forem/forem/pull/13007 I think the distinct formatting makes sense, as this is used for our runtime content filter not "classic" CSS.

**Comment by citizen428** [comment]
This looks good. Just one question: should there also be a data update script to remove the feature flag from all instances?

**Comment by fdoxyz** [comment]
> Just one question: should there also be a data update script to remove the feature flag from all instances?

Makes a lot of sense, totally missed that! Thanks @citizen428

File: app/views/layouts/application.html.erb
Comment (fdoxyz) [reply app/views/layouts/application.html.erb:107]: That's correct, the definition comes from [RFC 27 (Implementation)](https://github.com/forem/rfcs/blob/315ada9bbd6a033f831a33286b0066d7e8553c74/text/0027-runtime-content-filter.md#designimplementation).

We could also consider adding an extra word to them (i.e. `Browser-iOS-filter` or `Browser-iOS-only`). That would make them a bit more self-explanatory (for those without context), despite being more verbose it _could_ be valuable.

I don't have strong opinions on this though (I'm okay with leaving it as-is) but I'm open to revisiting it anytime 🙂

**Comment by fdoxyz** [comment]
Marked it as Draft to avoid accidental merges until we are ready to ship this

File: app/views/layouts/application.html.erb
Comment (citizen428) [reply app/views/layouts/application.html.erb:107]: Same as you, I don't feel strongly about it but I do like how declarative and self-explanatory  `Browser-iOS-only` is.

**Comment by ludwiczakpawel** [comment]
@fdoxyz is there a way to target specific view like a post view? the positioning issue is caused by action bar ("like", "super like", etc.) being sticky to the bottom edge of the screen but now the action bar is being "overlayed" with runtime banner. so if we could target that one view with CSS (body class?) we could avoid this issue.

File: app/views/layouts/application.html.erb
Comment (Ridhwana) [reply app/views/layouts/application.html.erb:107]: + 1 on `Browser-iOS-only` for people that are encountering it for the first time

**Review by rhymes — APPROVED** [review]
🔥

File: app/views/layouts/application.html.erb
Comment (vaidehijoshi) [reply app/views/layouts/application.html.erb:107]: Thanks for the explanation, folks! (Apparently even though I _reviewed_ that RFC I promptly forgot about why we named that class that way 😛)

I always will be the one to vote for explicit naming when possible, so `Browser-iOS-only` definitely seems more clear and self-explanatory 👍🏽

File: app/assets/javascripts/initializers/initializeRuntimeBanner.js
Comment (fdoxyz) [reply app/assets/javascripts/initializers/initializeRuntimeBanner.js:59]: Confirmed

File: app/views/layouts/application.html.erb
Comment (fdoxyz) [reply app/views/layouts/application.html.erb]: To confirm here: I updated these classes to include the `-only` suffix 😃

**Comment by fdoxyz** [comment]
@ludwiczakpawel I just pushed in changes that take this into account. I was able to re-use the `data-current-page='stories-show'` data attribute in the DOM to avoid the banner to overlay the reactions (only in small screens bc iPads show the reactions in a "left sidebar").

We'll merge this when Marketing promotion is shipped. In the mean time, anyone and everyone is welcome to test this again (if they want to double check things) 🙂

**Review by citizen428 — APPROVED** [review]

**Comment by lisasy** [comment]
Hi @fdoxyz, I am looking at this PR and I was wondering what your timeline is for getting this merged. Now that we've updated our app icon and have a crisper idea on how to extend the brand system into our product UI, I recommend making minor adjustments to the copy, padding, color, and possibly graphic to make this banner look a tad more welcoming — is this something we can do? And when would be most ideal?

**Comment by fdoxyz** [comment]
Hi @lisasy, we were thinking of merging tomorrow but we can hold it until we make those changes. Totally open to what you consider best though

**Comment by lisasy** [comment]
We can update the UI with color, icon, wording, and spacing to make this just one bit more inviting, showcase our brand, and connect this to the Forem app:

2021-05-27_Upsell-Banner


https://www.figma.com/file/1Xaf6Om9lxMlBJWdsbyran/Banner-Upsell?node-id=2%3A112

**Comment by fdoxyz** [comment]
@lisasy I just pushed in the changes that implement the design in that Figma link.

If you want to check them out locally I would suggest removing [this `Browser-iOS-only` class](https://github.com/forem/forem/pull/13662/files#diff-f43fe075643e681b2c01c2f853bb0c4299d135b47fcbd4da96890d521c49e3ebR107) after you checkout this branch. That will make it easier to see the banner on your browser, otherwise you'll need to use a real iOS device to see it.

Screen Shot 2021-05-27 at 17 12 37

File: app/assets/stylesheets/runtime-banner.scss
Comment (lisasy) [inline app/assets/stylesheets/runtime-banner.scss:14]:
```suggestion
  padding: var(--su-3)
```

File: app/assets/stylesheets/runtime-banner.scss
Comment (lisasy) [inline app/assets/stylesheets/runtime-banner.scss:37]:
```suggestion
      margin-bottom: var(--su-1);
```

File: app/views/layouts/application.html.erb
Comment (lisasy) [inline app/views/layouts/application.html.erb:116]:
```suggestion
        <button type="button" class="runtime-banner__dismiss crayons-btn crayons-btn--ghost crayons-btn--icon crayons-btn--inverted crayons-btn--s">
```

**Review by lisasy — APPROVED** [review]
I left minor CSS comments but otherwise this looks good. Thanks for the quick turnaround on styling updates! :)

**Pull request merged**
